tahrir/endpoints/users.py

```diff
@@ -44,3 +47,142 @@ def search_users_by_string(search_string: str):
     }
 
     return jsonify(result)
+
+
+@bp.route("/api/users/<string:user_id>", methods=["GET"])
+def get_user_by_id(user_id: str):
+    """Endpoint to fetch the user based on matching id."""
+
+    user = get_person(user_id)
+
+    if not user:
+        abort(404, f"No such user {user_id!r}")
+
+    if user.opt_out and user.email != g.oidc_user.email:
+        abort(404, f"User {user_id!r} has opted out.")
+
+    # Serialize invitations
+    invitations = [
+        {
+            "id": inv.id,
+            "badge_id": inv.badge_id,
+            "expires_on": inv.expires_on.isoformat(),
+            "created_on": inv.created_on.isoformat(),
+        }
+        for inv in g.tahrirdb.get_invitations(user.id)
+        if inv.expires_on > datetime.now()
+    ]
+
+    # Get badge info using utility function
+    badges_info = get_user_badge_info(user)
+
+    return jsonify(
+        {
+            "user": {
+                "id": user.id,
+                "nickname": user.nickname,
+                "email": user.email,
+                "avatar": user.avatar,
```

Comment:
```suggestion
                "mail": user.avatar,
```

As the email can be constructed using the `username`, there is no point in passing it. We only need the `avatar` here in order to be able to retrieve their portraits from Libravatar.

Maybe in the future, instead of passing down mails like these, we should hash them and pass them down to ensure that we do not serve email addresses to spammers on a silver platter.